vendor/github.com/mongodb/mongo-tools-common/db/buffered_bulk.go

```diff
@@ -67,30 +69,43 @@ func (bb *BufferedBulkInserter) Insert(doc interface{}) error {
 	return bb.InsertRaw(rawBytes)
 }
 
-// InsertRaw adds a document, represented as raw bson bytes, to the buffer for bulk insertion. If the buffer is full,
-// the bulk insert is made, returning any error that occurs.
-func (bb *BufferedBulkInserter) InsertRaw(rawBytes []byte) (err error) {
-	// flush if we are full
-	//
-	// XXX With OP_MSG the limit is larger; MaxBSONSize shouldn't be hard
-	// coded, it should be based on the server's ismaster response.
-	if bb.docCount >= bb.docLimit || bb.byteCount+len(rawBytes) > MaxBSONSize {
```

Comment:
do we not need to worry about this case anymore where we have too many bytes?

Reply:
The driver's bulk write API should take care of batch splitting and ensuring each batch is appropriately sized. In theory, we could get rid of `docLimit` too and do one big bulk write at the end of mongoimport. I chose to leave in `docLimit` because this way, if there are unrecoverable errors, they'll be caught earlier rather than at the very end.